userspace/libsinsp/runc.cpp

```diff
@@ -55,12 +73,14 @@ bool match_one_container_id(const std::string &cgroup,
 		return false;
 	}
 
-	if(end_pos - start_pos != CONTAINER_ID_LENGTH) {
+	// In some container runtimes the container the container id is not
+	// necessarly CONTAINER_ID_LENGTH long and can be arbitrarly defined.
+	// To keep it simple we only discard the container id > of CONTAINER_ID_LENGTH.
+	if(end_pos - start_pos > CONTAINER_ID_LENGTH || end_pos - start_pos == 0) {
```

Comment:
So, we only accept container_id whose length is <= CONTAINER_ID_LENGTH right?

Reply:
Below in this function we call
```
container_id = cgroup.substr(start_pos, REPORTED_CONTAINER_ID_LENGTH);
```
What if cgroup length is less than REPORTED_CONTAINER_ID_LENGTH? It would throw an `out_of_range` exception.

Reply:
ops! good point!